dashboards-observability/public/components/metrics/redux/slices/metrics_slice.ts

```diff
@@ -3,70 +3,85 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { createSlice } from '@reduxjs/toolkit';
-import { forEach } from 'lodash';
-import {
-  SELECTED_METRICS,
-  RECENTLY_CREATED_METRICS,
-  AVAILABLE_METRICS,
-  REDUX_SLICE_METRICS,
-} from '../../../../../common/constants/metrics';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { REDUX_SLICE_METRICS } from '../../../../../common/constants/metrics';
+import { pplServiceRequestor, getVisualizations } from '../../helpers/utils';
+import PPLService from '../../../../services/requests/ppl';
 
-const METRIC_NAMES_PPL = 'source = prometheus.information_schema.tables';
-import { metricNamesTablePPL } from './mockMetrics';
+const initialState = {
+  pplService: PPLService,
+  metrics: [],
+  selected: [],
+};
 
-const initialMetrics = {
-  [SELECTED_METRICS]: [],
-  [RECENTLY_CREATED_METRICS]: [],
-  [AVAILABLE_METRICS]: metricNamesTablePPL.datarows,
+export const loadMetrics = createAsyncThunk('metrics/loadData', async (services: any) => {
+  const { http, pplService } = services;
+  const customData = await fetchCustomMetrics(http);
+  const remoteData = await fetchRemoteMetrics(pplService);
+
+  return Promise.all([customData, ...remoteData]).then((datasets) => datasets.flat());
+});
+
+const fetchCustomMetrics = async (http: any) => {
+  const dataSet = await getVisualizations(http);
+
+  const normalizedData = dataSet.visualizations.map((obj: any) => ({
+    id: obj.id,
+    name: obj.name,
+    catalog: 'CUSTOM_METRICS',
+    type: obj.type,
+  }));
+  return normalizedData;
 };
 
-const initialState = {
-  metrics: { ...initialMetrics },
+const fetchRemoteMetrics = async (pplService: any) => {
+  const dataSet = [];
+  const catalogs = await pplServiceRequestor(
+    pplService,
+    'show catalogs | where CONNECTOR_TYPE="PROMETHEUS" | fields CATALOG_NAME'
```

Comment:
Let's move this query to `common/constants/metrics.ts`

Reply:
We can do this in future PRs

Reply:
moved